modules/version.nix

```diff
@@ -11,8 +11,16 @@ with lib;
   options = {
 
     system.stateVersion = mkOption {
-      type = types.enum [ "19.09" "20.03" "20.09" "21.05" "21.11" "22.05" ];
-      default = "19.09";
+      type = types.enum [
+        "19.09"
+        "20.03"
+        "20.09"
+        "21.05"
+        "21.11"
+        "22.05"
+        "22.11"
+        "23.05"
```

Comment:
I would recommend to make this option not an enum. This makes updated in the future easier and frankly out should not matter of I set this to 22.12.

home-manager is doing the same and it is annoying with every release and only created extra work.

Reply:
Why should it be possible to allow arbitrary values here? Setting stateVersion ="foo" doesn't make any sense and is prone to errors because we can not use this value at all.

Reply:
Any value that matches the regex ``\d\d\.\d\d`` should work.